test/cpp/tensorexpr/test_kernel.cpp

```diff
@@ -261,5 +262,164 @@ void testKernel_4() {
   }
 }
 
+namespace {
+
+std::string dtypeConstant(ScalarType scalar_type) {
+  if (scalar_type == ScalarType::None) {
+    return "None = prim::Constant()";
+  } else {
+    TemplateEnv env_dtype;
+    env_dtype.d("scalar_type", static_cast<int>(scalar_type));
+    return format("int = prim::Constant[value=${scalar_type}]()", env_dtype);
+  }
+}
+
+at::Tensor iotaTensor(IntArrayRef sizes, const at::TensorOptions& options) {
+  int64_t numel = std::accumulate(
+      sizes.begin(), sizes.end(), 1, std::multiplies<int64_t>());
+  std::vector<float> values(numel);
+  std::iota(values.begin(), values.end(), 0);
+  auto a = at::tensor(values, options);
+  return a.reshape(sizes);
+}
+
+} // namespace
+
+void testKernelSumAllAxes() {
+  // Test lowering of sum on all axes.
+  const auto graph_template = R"IR(
+      graph(%0 : Float(5:3,3:1, device=cpu)):
+        %1 : ${dtype}
+        %2 : Float() = aten::sum(%0, %1)
+        return (%2))IR";
+  auto a = iotaTensor({5, 3}, TensorOptions(kCPU).dtype(at::kFloat));
+
+  for (auto scalar_type : {ScalarType::None, ScalarType::Double}) {
+    KernelScope kernel_scope;
+    TemplateEnv env;
+    env.s("dtype", dtypeConstant(scalar_type));
+    const auto graph_string = format(graph_template, env);
+
+    auto graph = std::make_shared<Graph>();
+    parseIR(graph_string, &*graph);
+
+    auto o = at::empty({}, TensorOptions(kCPU));
+    c10::optional<c10::ScalarType> dtype;
+    if (scalar_type != ScalarType::None) {
+      dtype = static_cast<c10::ScalarType>(scalar_type);
+    }
+    auto ref = a.sum(/*dtype=*/dtype);
+    TensorExprKernel k(graph);
+    std::vector<at::Tensor> inputs = {a};
+    Stmt* s = k.getCodeGenStmt();
+    // TODO: verify stmt
+
+    std::vector<IValue> stack = fmap<IValue>(inputs);
+    k.run(stack);
+    o = stack[0].toTensor();
+    ASSERT_EQ(o.sizes(), ref.sizes());
+    ASSERT_TRUE(at::allclose(o, ref));
+  }
+}
+
+void testKernelSumOneAxis() {
+  // Test lowering of sum on one axis.
+  const auto graph_template = R"IR(
+      graph(%0 : Float(5:3,3:1, device=cpu)):
+        %1 : int = prim::Constant[value=${dim}]()
+        %2 : int[] = prim::ListConstruct(%1)
+        %3 : bool = prim::Constant[value=${keepdim}]()
+        %4 : ${dtype}
+        %5 : Float() = aten::sum(%0, %2, %3, %4)
```

Comment:
The output shape seems to be wrong if we're summing only one axis (and it will also depend on the `${keepdim}`. You could use `Tensor` here, but that would require adding supporting `aten::sum` in the NNC shape inference (which shouldn't be very difficult).

Reply:
Just `Tensor` worked, probably because we don't have fusion with any other operator in this PR. I'll have to add the shape inference in my next PRs, yes.

Reply:
Yeah, it probably would've failed if we tried to use the output as an input of some other node.